File: backend/src/main/java/edu/cit/cabatana/doughlycrumbl/features/rating/OrderRating.java
package edu.cit.cabatana.doughlycrumbl.features.rating;

import edu.cit.cabatana.doughlycrumbl.features.order.Order;
import edu.cit.cabatana.doughlycrumbl.features.user.User;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "order_ratings", uniqueConstraints = {
        @UniqueConstraint(columnNames = {"order_id", "user_id"})
})
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class OrderRating {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "order_id", nullable = false)
    private Order order;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private Integer rating;

    @Column(columnDefinition = "TEXT")
    private String comment;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
}

File: mobile/app/src/main/java/com/example/mobile/model/OrderRating.kt
package com.example.mobile.model

import com.google.gson.annotations.SerializedName

data class OrderRating(
    @SerializedName("id") val id: Long = 0,
    @SerializedName("orderId") val orderId: Long = 0,
    @SerializedName("rating") val rating: Int = 0,
    @SerializedName("comment") val comment: String? = null,
    @SerializedName("createdAt") val createdAt: String? = null
)

data class OrderRatingRequest(
    @SerializedName("rating") val rating: Int,
    @SerializedName("comment") val comment: String? = null
)
